aesara/tensor/basic.py

```diff
@@ -562,6 +562,25 @@ def grad(self, inp, grads):
 
         raise NotImplementedError("grad not implemented for complex dtypes")
 
+    def c_code(self, node, name, inputs, outputs, sub):
+        (x,) = inputs
+        (z,) = outputs
+        fail = sub["fail"]
+
+        return (
+            """
+            %(z)s = (PyArrayObject*)PyArray_FromScalar(py_%(x)s, NULL);
+            if(py_%(z)s == NULL){
+                %(fail)s;
+            }
+            Py_XINCREF(%(z)s);
```

Comment:
I am always confused by the reference counting, but do we need this? `ScalarFromTensor` doesn't do it. Is it because there the output is not an array?

Reply:
Yes, we need the `INCREF` there.

Reply:
> Is it because there the output is not an array there?

After the call to `PyArray_FromScalar`, the returned object has a reference count of one, which I believe is subsequently decremented (see the code in https://github.com/aesara-devs/aesara/pull/1124#issuecomment-1218608975), so we need the extra increment to keep it from getting freed.

Really, my comment above is pointing out how the C code generated for scalars is perhaps unreasonably specialized for only certain use cases.